maven-release-manager/src/main/java/org/apache/maven/shared/release/transform/jdom2/JDomProperties.java

```diff
@@ -51,10 +51,13 @@ public JDomProperties(Element properties) {
     }
 
     @Override
-    public synchronized Object setProperty(String key, String value) {
-        Element property = properties.getChild(key, properties.getNamespace());
+    public synchronized Object put(Object key, Object value) {
+        Element property = properties.getChild((String) key, properties.getNamespace());
 
-        JDomUtils.rewriteValue(property, value);
+        if (property == null) {
+            property = new Element((String) key, properties.getNamespace());
```

Comment:
I do not understand this logic. Why is this a `put` here?

Reply:
Because the class extends `Properties` which extends `Map`, hence, it should override `put` instead of `setProperty`, so it can be used as `Map`